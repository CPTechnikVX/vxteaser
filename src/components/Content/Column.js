import React      from 'react';
import Constants  from '../../utils/Constants';
import PropTypes  from 'prop-types';
import classnames from 'classnames';

/**
 * Column element for the grid
 */
export default class Column extends React.PureComponent {
    render() {
        const {children, modifier, width} = this.props;
        const styleObj                    = {
            width: width ? width : 'auto',
        };
        const classList                   = [Constants.ClassName.Column];

        if (modifier) {
            classList.push(modifier);
        }

        return <div className={classnames(classList)} style={styleObj} data-testid="column">{children}</div>;
    }
}

Column.propTypes = {
    /** @ignore */
    children: PropTypes.node,
    modifier: PropTypes.string,
    width:    PropTypes.string,

};
