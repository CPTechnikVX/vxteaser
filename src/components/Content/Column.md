```html
<Col width="20%">This is a column of 20% width</Col>
```

```html
<Col>This is a column with an automatic width</Col>
```
