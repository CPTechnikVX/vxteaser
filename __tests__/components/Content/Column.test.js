import React    from 'react';
import Column   from '../../../src/components/Content/Column';
import {render} from '@testing-library/react';

describe('render column', () => {
    test('empty', () => {
        const {container} = render(
                <Column width="33%" />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test('with content', () => {
        const {container} = render(
                <Column width="33%">Column content</Column>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
