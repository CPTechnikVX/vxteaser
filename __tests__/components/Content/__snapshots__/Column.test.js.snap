// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render column empty 1`] = `
<div
  class="vxteaser-col"
  data-testid="column"
  style="width: 33%;"
/>
`;

exports[`render column with content 1`] = `
<div
  class="vxteaser-col"
  data-testid="column"
  style="width: 33%;"
>
  Column content
</div>
`;
